pkg/apiserver/certificate/cacert_controller.go

```diff
@@ -130,39 +137,50 @@ func (c *CACertController) syncCACert() error {
 // syncMutatingWebhooks updates the CABundle of the MutatingWebhookConfiguration backed by antrea-controller.
 func (c *CACertController) syncMutatingWebhooks(caCert []byte) error {
 	klog.Info("Syncing CA certificate with MutatingWebhookConfigurations")
-	for _, name := range c.caConfig.MutationWebhooks {
-		mWebhook, err := c.client.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
-		if err != nil {
-			return fmt.Errorf("error getting MutatingWebhookConfiguration %s: %v", name, err)
-		}
-		err = c.patchWebhookWithCACert(mWebhook, caCert)
-		if err != nil {
-			return fmt.Errorf("error updating antrea CA cert of MutatingWebhookConfiguration %s: %v", name, err)
-		}
+	mWebhooks, err := c.client.AdmissionregistrationV1().MutatingWebhookConfigurations().List(context.TODO(), antreaListOption)
+	if err != nil {
+		return fmt.Errorf("error listing antrea MutatingWebhookConfiguration: %v", err)
 	}
-	for _, name := range c.caConfig.OptionalMutationWebhooks {
-		mWebhook, err := c.client.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
-		if err != nil {
-			if errors.IsNotFound(err) {
-				klog.V(2).Infof("Optional mutation webhook %s not found, skipping its update", name)
-				continue
+	supportedMutatingWebhooks := sets.NewString(c.caConfig.MutationWebhooks...)
+	supportedOptionalWebhooks := sets.NewString(c.caConfig.OptionalMutationWebhooks...)
+	unSupporteWebhooks := sets.NewString()
+	for i := range mWebhooks.Items {
```

Comment:
I feel it would become complexer to handle the webhooks. We should use either discovery mechanism or static list to fetch the webhooks need to inject CA, but not both, which just introduces two sources of truth and unnecessary mismatching handling.

A simpler solution in my mind is:
1. We add one extra label to Webhook resources to indicate which component is the backend of the webhook, like "served-by=antrea-controller", "backed-by=antrea-controller", or "backend-service=antrea-controller".
2. Remove "caConfig.MutationWebhooks..." and "caConfig.OptionalMutationWebhooks...", and add one label selector `caConfig.MutationWebhookSelector *metav1.LabelSelector`, when it's set, CACertController should fetch webhooks selected by it and inject CA to them. And antrea-controller and antrea-mc-controller could pass different labels to fetch their own webhoosk.

Then we don't have to duplicate and hard code the webhook list in code and the deployment manifest is the only source of truth of which webhooks should be injected.

It applies to other webhooks and APIServices.